extensions/cms/src/cmsUtils.ts

```diff
@@ -70,49 +104,70 @@ export class CmsUtils {
 		if (!ownerUri) {
 			// Make a connection if it's not already connected
-			await azdata.connection.connect(Utils.toConnectionProfile(connection), false, false).then(async (result) => {
-				ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
-			});
+			let initialConnectionProfile = CmsUtils.getConnectionProfile(connection);
+			let result = await azdata.connection.connect(initialConnectionProfile, false, false);
+			ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+			// If the ownerUri is still undefined, then open a connection dialog with the connection
+			if (!ownerUri) {
+				try {
+					let result = await this.connection(initialConnectionProfile);
+					if (result) {
+						ownerUri = await azdata.connection.getUriForConnection(result.connectionId);
+						connection = result;
+					}
+				} catch (error) {
+					// cancel pressed on connection dialog
+					const errorText = localize('cms.error.cancelConnectionDialog', 'The server is disconnected. Please connect to the server to continue.');
+					return Promise.reject(error ? new Error(error.message) : errorText);
+				}
+			}
 		}
-		return provider.createCmsServer(name, description, connection, ownerUri).then((result) => {
+		try {
+			let result = await provider.createCmsServer(name, description, connection, ownerUri);
 			if (result) {
-				return Promise.resolve(result);
-			} else {
-				return Promise.reject(null);
+				const createCmsResult: CreateCmsResult = {
+					listRegisteredServersResult: result,
+					connection: connection
+				};
+				return Promise.resolve(createCmsResult);
 			}
-		});
+		} catch (errorString) {
+			const error = new Error(errorString.message);
+			return Promise.reject(error);
+		}
 	}
 
-	public async deleteCmsServer(cmsServer: any): Promise<void> {
+	public async deleteCmsServer(cmsServerName: string, connection: azdata.connection.Connection): Promise<void> {
 		let config = this.getConfiguration();
 		if (config && config.servers) {
 			let newServers = config.servers.filter((cachedServer) => {
-				return cachedServer.name !== cmsServer;
+				return cachedServer.name !== cmsServerName;
 			});
 			await this.setConfiguration(newServers);
 			this._registeredCmsServers = this._registeredCmsServers.filter((cachedServer) => {
-				return cachedServer.name !== cmsServer;
+				return cachedServer.name !== cmsServerName;
 			});
 		}
+		if (connection.options.authenticationType === sqlLoginAuthType && connection.options.savePassword) {
+			this._credentialProvider.deleteCredential(connection.options.user);
+		}
 	}
 
-	public cacheRegisteredCmsServer(name: string, description: string, ownerUri: string, connection: azdata.connection.Connection): void {
+	public async cacheRegisteredCmsServer(name: string, description: string, ownerUri: string, connection: azdata.connection.Connection): Promise<void> {
 		if (!this._registeredCmsServers) {
 			this._registeredCmsServers = [];
 		}
```

Comment:
Just get rid of this and initialize the var to an empty array to start with so we don't have to do this check every single time